src/rules/expect-expect.ts

```diff
@@ -7,12 +7,12 @@ import {
   AST_NODE_TYPES,
   TSESTree,
 } from '@typescript-eslint/experimental-utils';
-import micromatch from 'micromatch';
 import {
   TestCaseName,
   createRule,
   getNodeName,
   getTestCallExpressionsFromDeclaredVariables,
+  nodeNameMatch,
```

Comment:
I think it would better to have this in the rule rather than in `utils`, since it's only used here - we can easily move it out later if we find further use for it, but right now it's API is lightly coupled to `assertFunctionNames`.

Also, bit of a nit but I think something like `matchesAssertFunctionName` would be a better name, to give the impression it's a check.